### PR TITLE
Update: return type error in `valid-jsdoc` rule (fixes #4443)

### LINKED ISSUE #4443: valid-jsdoc: Throws exception when @returns has no type
Given `test.js`:

```
/**
 * @returns something
 */
function foo() {

}
```

When I run `eslint --rule 'valid-jsdoc: 1' test.js`:

Then:

```
node_modules/eslint/lib/rules/valid-jsdoc.js:69
        return tag.type.name === "void" || tag.type.type === "UndefinedLiteral";
                       ^

TypeError: Cannot read property 'name' of null
```

If the JSDoc comment instead reads:

```
/**
 * @returns {string} something
 */
```

Then ESLint runs as expected with no output. 

Run with ESLint 1.9.0, Node v4.1.2 and NPM 3.4.0.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/28280864-valid-jsdoc-throws-exception-when-returns-has-no-type?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by gyandeeps**

### CODE REVIEW

File: tests/lib/rules/valid-jsdoc.js [hunk 1]
```diff
@@ -429,6 +429,14 @@ ruleTester.run("valid-jsdoc", rule, {
                 type: "Block"
             }]
         },
+        {
```
Comment (ilyavolodin) [inline tests/lib/rules/valid-jsdoc.js]: Can you add a test with `@returns void` too?

Comment (gyandeeps) [inline tests/lib/rules/valid-jsdoc.js]: Done

Comment (ilyavolodin) [inline tests/lib/rules/valid-jsdoc.js]: Sorry, maybe I'm missing something, but based on the issue this is fixing, `@returns void` shouldn't cause an error. As far as I can tell, @nzakas said that type is not required for `return` statement.

Comment (gyandeeps) [inline tests/lib/rules/valid-jsdoc.js]: I also thought about what you saying. I think to correctly manage that we need an option `requireReturnType` by default set to `true`. This way we can do non-breaking change.

Comment (nzakas) [inline tests/lib/rules/valid-jsdoc.js]: Good idea

Comment (gyandeeps) [inline tests/lib/rules/valid-jsdoc.js]: :+1: 
Will make that change later today.

**Comment by nzakas** [comment]
LGTM

**Pull request merged**